go/protocol/keybase1/extras.go

```diff
@@ -988,3 +996,69 @@ func (u UserPlusAllKeys) IsOlderThan(v UserPlusAllKeys) bool {
 	}
 	return false
 }
+
+func (ut UserOrTeamID) String() string {
+	return string(ut)
+}
+
+func (ut UserOrTeamID) ToBytes() []byte {
+	b, err := hex.DecodeString(string(ut))
+	if err != nil {
+		return nil
+	}
+	return b
+}
+
+func (ut UserOrTeamID) IsNil() bool {
+	return len(ut) == 0
+}
+
+func (ut UserOrTeamID) Exists() bool {
+	return !ut.IsNil()
+}
+
+func (ut UserOrTeamID) Equal(v UserOrTeamID) bool {
+	return ut == v
+}
+
+func (ut UserOrTeamID) NotEqual(v UserOrTeamID) bool {
+	return !ut.Equal(v)
+}
+
+func (ut UserOrTeamID) Less(v UserOrTeamID) bool {
+	return ut < v
+}
+
+func (ut UserOrTeamID) AsUser() (UID, error) {
```

Comment:
For conciseness, would it make sense to add a `AsUserOrBust` and `AsTeamOrBust`? If we know it's a user or a team, it'd be nice to not have to check the error.

(Or rename this to `TryAsUser` and have `AsUser` one bust if it's not a user, for consistency with `AsUserOrTeam`